test/cobalt/experiments/test_rag_quality.py

```
@@ -0,0 +1,246 @@
+            context = ' '.join(rag_response['retrieved_docs'])
+            answer = rag_response['answer']
+
+            evaluation = evaluate_grounding(answer, context)
+            scores.append(evaluation['score'])
+
+            assert evaluation['score'] >= 0.4, f'Answer for "{item["query"]}" is not well grounded: {evaluation["reasoning"]}'
+
+        avg_score = sum(scores) / len(scores)
+        assert avg_score >= 0.5, f'Average grounding score {avg_score:.2f} is below threshold 0.5'
+
+    def test_hallucination_detection(self, mock_rocketride_client, sample_rag_dataset):
```

Ruff (0.15.7)

[warning] Unused method argument: `mock_rocketride_client`

(ARG002)